vars/promoteRepos.groovy

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+
+ /**
+ * Library to sign and promote repositories to artifacts.opensearch.org
+@param Map[jobName] <required> - Name of the distribution build workflow on Jenkins that build the artifacts
+@param Map[buildNumber] <required> - The build number of the artifacts in above build workflow
+@param Map[distributionRepoType] <required> - distribution repository type, e.g. yum / apt
+@param Map[manifest] <optional> - input manifest of the corresponding OpenSearch/OpenSearch-Dashboards product version (e.g. 2.0.0)
+ */
+
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+
+    String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+    def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
+
+    String filename = inputManifest.build.getFilename()
+    String jobname = args.jobName
+
+    String buildnumber = args.buildNumber ?: 'none'
+    if (buildnumber == 'none') {
+        println('User did not enter build number in jenkins parameter, exit 1')
+        System.exit(1)
+    }
+
+    def distTypeMap = [
+        "yum": "rpm",
+        "apt": "deb",
+    ]
+    String repoType = args.distributionRepoType
+    String distType = distTypeMap["${repoType}"]
+    String version = inputManifest.build.version
+    String majorVersion = version.tokenize('.')[0]
+    String repoVersion = majorVersion + '.x'
+    String qualifier = inputManifest.build.qualifier ? '-' + inputManifest.build.qualifier : ''
+    String revision = version + qualifier
+    println("Product: ${filename}")
+    println("Build Number: ${buildnumber}")
+    println("Input Manifest: ${manifest}")
+    println("Revision: ${revision}")
+    println("Major Version: ${majorVersion}")
+    println("Repo Type: ${repoType}")
+    println("Dist Type: ${distType}")
+    println("Repo Version: ${repoVersion}")
```

Comment:
Is this really required?

Reply:
Yes, it is easier to understand what is happening and which one is being uploaded.
More output is better considering the complexity of this lib.